python_modules/dagster/dagster/_core/definitions/asset_graph_differ.py

```diff
@@ -28,6 +27,79 @@ class AssetDefinitionChangeType(Enum):
     REMOVED = "REMOVED"
 
 
+@whitelist_for_serdes
+@record
+class AssetDefinitionChangeDetailNew:
+    change_type: AssetDefinitionChangeType = AssetDefinitionChangeType.NEW
+
+
+@whitelist_for_serdes
+@record
+class AssetDefinitionChangeDetailCodeVersion:
+    change_type: AssetDefinitionChangeType = AssetDefinitionChangeType.CODE_VERSION
+    base_value: Optional[str] = None
+    branch_value: Optional[str] = None
+
+
+@whitelist_for_serdes
+@record
+class AssetDefinitionChangeDetailDependencies:
+    change_type: AssetDefinitionChangeType = AssetDefinitionChangeType.DEPENDENCIES
+    added: Sequence[AssetKey] = []
+    changed: Sequence[AssetKey] = []
+    removed: Sequence[AssetKey] = []
+
+
+@whitelist_for_serdes
+@record
+class AssetDefinitionChangeDetailPartitionsDefinition:
+    change_type: AssetDefinitionChangeType = AssetDefinitionChangeType.PARTITIONS_DEFINITION
+    base_value: Optional[str] = None
+    branch_value: Optional[str] = None
+
+
+@whitelist_for_serdes
+@record
+class AssetDefinitionChangeDetailTags:
+    change_type: AssetDefinitionChangeType = AssetDefinitionChangeType.TAGS
+    added: Sequence[str] = []
+    changed: Sequence[str] = []
+    removed: Sequence[str] = []
+
+
+@whitelist_for_serdes
+@record
+class AssetDefinitionChangeDetailMetadata:
+    change_type: AssetDefinitionChangeType = AssetDefinitionChangeType.METADATA
+    added: Sequence[str] = []
+    changed: Sequence[str] = []
+    removed: Sequence[str] = []
+
+
+@whitelist_for_serdes
+@record
+class AssetDefinitionChangeDetailRemoved:
+    change_type: AssetDefinitionChangeType = AssetDefinitionChangeType.REMOVED
+
+
+"""Represents the diff information for changes between base and branch assets."""
```

Comment:
Would it make sense to instead represent this common type using a base class that the change detail classes extend from?

Reply:
I think that makes more sense. Will switch to using a base class.